### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by aliabbani**
## Blog App - processing data in models
In this milestone I:
- Create model classes for all entities as shown in the ERD diagram.
- Set up associations between models.
- Create a method in the user model that returns the 3 most recent posts for a given user.
- Create a method in the post model that updates the posts counter for a user.
- Create a method in the post model which returns the 5 most recent comments for a given post.
-  Create a method in the comment model that updates the comments counter for a post.
-  Create a method in the like model that updates the likes counter for a post.
- Create a method called current_user in the ApplicationController that returns the first user from a database. Without worrying about proper authentication.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -6,4 +6,5 @@ class User < ApplicationRecord
   def recent_post
     posts.order('created_at Desc').limit(3)
   end
+
```
Comment (dasileker) [inline app/models/user.rb]: - [x] The `User.rb`  association should have the foreign keys with ID.

File: app/models/post.rb [hunk 2]
```diff
@@ -10,7 +10,6 @@ def recent_comments
   after_save :update_posts_counter
```
Comment (dasileker) [inline app/models/post.rb]: - [x] Also in line number 4 the association should be attributed to the User class name.

File: app/models/like.rb [hunk 3]
```diff
@@ -5,7 +5,6 @@ class Like < ApplicationRecord
   after_save :update_likes_counter
```
Comment (dasileker) [inline app/models/like.rb:5]: - [x] Same thing for line 1.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi ,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

- Please make changes on files because right now I can't see any changes made.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (aliabbani) [reply app/models/user.rb]: Done :white_check_mark:

File: app/models/post.rb (hunk 2 above)
Comment (aliabbani) [reply app/models/post.rb]: Done :white_check_mark:

File: app/models/like.rb (hunk 3 above)
Comment (aliabbani) [reply app/models/like.rb:5]: Done :white_check_mark:

**Review by Emmyn5600 — APPROVED** [review]
Hi @aliabbani 

Woo ow 👏, Your project is complete!  There is nothing else to say other than... it's time to merge it. 👏

Congratulations! 🎉 

##  Optional suggestion

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**